packages/material-ui/src/Table/Table.js

```diff
@@ -14,21 +15,15 @@ export const styles = theme => ({
   },
 });
 
-class Table extends React.Component {
-  getChildContext() {
-    // eslint-disable-line class-methods-use-this
-    return {
-      table: {
-        padding: this.props.padding,
-      },
-    };
-  }
+function Table(props) {
+  const { classes, className, component: Component, padding, ...other } = props;
+  const childContext = { padding };
```

Comment:
We could simply use the padding here to avoid child rerender every time the Provider rerenders since react uses strict equality for context values when checking if a consumer should rerender.